gr-audio/lib/portaudio/portaudio_impl.cc

```diff
@@ -14,82 +14,78 @@
 
 #include "portaudio_impl.h"
 #include <portaudio.h>
-#include <cstring>
+#include <string_view>
+#include <tuple>
+#include <vector>
 
 namespace gr {
 namespace audio {
 
 PaDeviceIndex pa_find_device_by_name(const char* name)
 {
-    int i;
-    int numDevices;
-    const PaDeviceInfo* pdi;
-    int len = strlen(name);
-    PaDeviceIndex result = paNoDevice;
-    numDevices = Pa_GetDeviceCount();
-    for (i = 0; i < numDevices; i++) {
-        pdi = Pa_GetDeviceInfo(i);
-        if (strncmp(name, pdi->name, len) == 0) {
```

Comment:
We were doing `str`**n**`cmp` for buffer overflow safety… using the `strlen` of an external string. So, nothing won here, honestly.